src/main/java/org/verdictdb/core/querying/ola/AsyncQueryExecutionPlan.java

```diff
@@ -327,7 +320,8 @@ public ExecutableNodeBase convertToProgressiveAgg(
 
     if (root instanceof AggExecutionNode) {
       // check if it contains at least one scrambled table.
-      if (doesContainScramble(root, scrambleMeta)) {
+      if (doesContainScramble(root, scrambleMeta) &&
```

Comment:
I am not sure I have clearly understood the structure. Should we check select list in method getScrambleMetaFrom()? StaticMetaData can only record which table contain which columns. For instance,
```
select avg(acct) 
from
(
  select c_acctbal as acct, l_extendedprice as price
  from customer, lineitem
) as vt1
```
Without checking select list in getScrambleMetaFrom(), I think we can only know vt1 is composed by customer and lineitem

Reply:
You can check the select list at this line `// also, populate newTableMeta for this source.`

For the example query you wrote, getScrambleMetaFrom() will first call itself again since the source is an instance of SelectQuery.

In the second call of getScrambleMetaFrom() with the inner query, the select list of vt1 is inserted into `newTableMeta` as { mostOuter: (acct, price) }, which is returned to the caller. "mostOuter" is used because vt1 is not visible in this scope. getScrambleMetaFrom() also record that "mostOuter" is a scrambled table. Initially, I included `ScrambleMetaSet` in the return type to record the information (that "mostOuter" is a scrambled table), but I guess simply using List<String> may be sufficient for this usage.

After obtaining { mostOuter: (acct, price) } for the inner query (i.e., vt1), getScrambleMetaFrom() now replaces the table name (i.e., mostOuter) with "vt1", which is the operation for `// replace the table names in StaticMetaData with source.alias`.

Let me know if this approach makes sense to you.

Reply:
I am still not very clear that after replacing the alias, we get {vt1:(acct, price)}. But how to check the belongingness of acct and price? I think the task should be finding out which column comes from which base table. Therefore, by checking the select list, we can get StaticMetaData like {customer:acct}, {lineitem:price}. So, we can know avg(acct) is the column does not involve scrambled table's column.

Reply:
Another problem is that for every base table in the from list, it will search all the base column in the select list(including the base column in the columnOp) to check whether this table appear in this select item. Will it be inefficient?

Reply:
Please work on https://github.com/mozafari/verdictdb/issues/245 then.